build-system/tasks/check-types.js

```diff
@@ -8,13 +8,29 @@ const {compileJison} = require('./compile-jison');
 const {cyan, green} = require('kleur/colors');
 const {execOrThrow} = require('../common/exec');
 const {log} = require('../common/logging');
+const {updateSubpackages} = require('../common/update-packages');
+const fastGlob = require('fast-glob');
+const path = require('path');
+
+/**
+ * Helper that updates build-system subpackages so their types can be verified.
+ * Skips npm checks during CI (already done while running each task).
+ */
+function updateBuildSystemSubpackages() {
+  const packageFiles = fastGlob.sync('build-system/tasks/*/package.json');
```

Comment:
Do we want just `/*/` or `/**/` here? ie. should it include subfolders? Guessing not but confirming

Reply:
There are `package.json` files deeper so I think we indeed need the recursive wildcard.

```suggestion
  const packageFiles = fastGlob.sync([
    'build-system/tasks/**/package.json',
    '!**/node_modules/**',
  ]);
```